managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2222,7 +2222,19 @@ public int getPendingAddEntriesCount() {
 
     @Override
     public Position getLastConfirmedEntry() {
-        return lastConfirmedEntry;
+        PositionImpl lastMessagePosition = lastConfirmedEntry;
```

Comment:
This could be done in initialize(), as once the last ledger has been loaded, you can set lastConfirmedEntry. lastConfirmedEntry only gets set with entry id as -1 in initializeBookKeeper() now. This could be removed if you do it in inititalize().